src/components/Inputs/input.module.scss

```diff
@@ -1438,6 +1447,114 @@
     &.input-stretch {
         width: 100%;
     }
+
+    &.isExpandable {
+        .expandable-wrapper {
+            position: relative;
+        }
+
+        .input-group {
+            position: absolute;
+            left: 0;
+            z-index: 10;
+
+            input {
+                width: 0;
+            }
+
+            .clear-icon-button {
+                right: -12%;
+            }
+        }
+
+        .expandable-thumb {
+            position: absolute;
+            left: -3px;
+        }
+
+        .overlay {
+            position: absolute;
+            left: 0;
+            top: 0;
+            height: 100%;
+            width: 200%;
+            z-index: -1;
+            background: $expandable-header-gradient;
+        }
+
+        &.expand-right {
+            .input-group,
+            .overlay {
+                left: unset;
+                right: 10px;
+            }
+            .overlay {
+                transform: rotate(180deg);
+            }
+
+            .expandable-wrapper {
+                margin-left: 10px;
+            }
+        }
+
+        input,
+        .input-group,
+        .overlay {
+            transition: all $motion-duration-fast $motion-easing-easeout;
+        }
+
+        // Set Rules
+        .input-group:hover > input,
+        input:focus,
+        input:not(:placeholder-shown) {
+            @each $size-name, $size-value in $input-size-map {
+                &.input-#{$size-name} {
+                    width: auto;
+                }
+            }
+
+            ~ .action-wrapper {
+                display: block;
+            }
+
+            ~ .expandable-thumb {
+                display: none;
+            }
+        }
+
+        // Unset Rules
+        @each $size-name, $size-value in $input-size-map {
+            &.input-#{$size-name} > .expandable-wrapper {
+                min-width: $size-value;
+                min-height: $size-value;
+            }
+
+            .input-group:not(:hover)
+                input.input-#{$size-name}:placeholder-shown {
+                min-width: $size-value;
+                padding: 0;
+                overflow: hidden;
+
+                // Internet Explorer 10–11
+                &:-ms-input-placeholder {
+                    font-size: 0;
+                }
+
+                // Edge
+                &::-ms-input-placeholder {
+                    font-size: 0;
+                }
+
+                &::placeholder {
+                    font-size: 0;
```

Comment:
font-size based change here is still resulting in overlap of the search text and icons during the animation/expansion.

Reply:
Increased the transition speed. It's bearly visible during expansion.

Reply:
It's still noticeable. You need to add a mask.